### ISSUE TITLE
[Magento2] Object manager return types not resolved propertly

### ISSUE THREAD

**Issue by Koc** [enriquepiatti/Magicento#238]
```php
$category = $this->_objectManager->create('Magento\Catalog\Model\Category');
$category->load($categoryId);
// type of category variable not resolved, no any autocomplete or ctrl+click
```

**Comment by enriquepiatti** [comment]
solved, will be available in next version (1.5+)
Thanks!!

**Comment by Koc** [comment]
@enriquepiatti awesome, helps a lot, thank you! One small thing: it doesn't work with `class` constant:

```php
$parentId = $this->objectManager
    ->create('Magento\ConfigurableProduct\Model\ResourceModel\Product\Type\Configurable')
    ->getParentIdsByChild($product->getId()); // type resolved

$parentId = $this->objectManager
    ->create(\Magento\ConfigurableProduct\Model\ResourceModel\Product\Type\Configurable:class)
    ->getParentIdsByChild($product->getId()); // unknown type
```

**Comment by enriquepiatti** [comment]
added support for that too (available in next release 1.6+)
